default/python/AI/freeorion_tools/_freeorion_tools.py

```diff
@@ -298,3 +298,52 @@ def dump_universe():
             dump_universe.last_dump < cur_turn):
         dump_universe.last_dump = cur_turn
         print fo.universe().dump()
+
+
+class LogLevelSwitcher(object):
```

Comment:
Maybe rewrite with [contextmanager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager), It will be much shorter.